rpcs3/rpcs3qt/cheat_manager.cpp

```diff
@@ -0,0 +1,660 @@
+
+		return 0;
+	}
+
+	success = true;
+
+	T ret_value = *vm::get_super_ptr<T>(offset);
+
+	if (did_we_pause)
+		Emu.Resume();
+
+	return ret_value;
+}
+
+template <typename T>
+bool cheat_engine::set_value(const u32 offset, const T value)
+{
+	if (!vm::check_addr(offset, vm::page_info_t::page_allocated | vm::page_info_t::page_writable))
+	{
+		return false;
+	}
+
+	*vm::get_super_ptr<T>(offset) = value;
+	return true;
+}
+
+cheat_manager_dialog::cheat_manager_dialog(QWidget* parent)
+    : QDialog(parent)
+{
+	setWindowTitle(tr("Cheat Manager"));
+	setObjectName("cheat_manager");
+	setMinimumSize(QSize(800, 400));
+
+	QVBoxLayout* main_layout = new QVBoxLayout();
+
+	tbl_cheats = new QTableWidget(this);
+	tbl_cheats->setSelectionMode(QAbstractItemView::SelectionMode::ExtendedSelection);
+	tbl_cheats->setSelectionBehavior(QAbstractItemView::SelectRows);
+	tbl_cheats->setContextMenuPolicy(Qt::CustomContextMenu);
+	tbl_cheats->setColumnCount(4);
+	tbl_cheats->setHorizontalHeaderLabels(QStringList() << tr("Game") << tr("Description") << tr("Type") << tr("Offset"));
+	main_layout->addWidget(tbl_cheats);
+
+	QHBoxLayout* btn_layout = new QHBoxLayout();
+	QLabel* lbl_value_final = new QLabel(tr("Current Value:"));
+	edt_value_final         = new QLineEdit();
+	btn_apply               = new QPushButton(tr("Apply"), this);
+	btn_apply->setEnabled(false);
+	btn_layout->addWidget(lbl_value_final);
+	btn_layout->addWidget(edt_value_final);
+	btn_layout->addWidget(btn_apply);
+	main_layout->addLayout(btn_layout);
+
+	QGroupBox* grp_add_cheat              = new QGroupBox(tr("Cheat Search"));
+	QVBoxLayout* grp_add_cheat_layout     = new QVBoxLayout();
+	QHBoxLayout* grp_add_cheat_sub_layout = new QHBoxLayout();
+	QPushButton* btn_new_search           = new QPushButton(tr("New Search"));
+	btn_filter_results                    = new QPushButton(tr("Filter Results"));
+	btn_filter_results->setEnabled(false);
+	edt_cheat_search_value = new QLineEdit();
+	cbx_cheat_search_type  = new QComboBox();
+	cbx_cheat_search_type->insertItems(0, cheat_types);
+	cbx_cheat_search_type->setCurrentIndex(signed_32_cheat);
+	grp_add_cheat_sub_layout->addWidget(btn_new_search);
+	grp_add_cheat_sub_layout->addWidget(btn_filter_results);
+	grp_add_cheat_sub_layout->addWidget(edt_cheat_search_value);
+	grp_add_cheat_sub_layout->addWidget(cbx_cheat_search_type);
+	grp_add_cheat_layout->addLayout(grp_add_cheat_sub_layout);
+	lst_search = new QListWidget(this);
+	lst_search->setSelectionMode(QAbstractItemView::SelectionMode::SingleSelection);
+	lst_search->setSelectionBehavior(QAbstractItemView::SelectRows);
+	lst_search->setContextMenuPolicy(Qt::CustomContextMenu);
+	grp_add_cheat_layout->addWidget(lst_search);
+	grp_add_cheat->setLayout(grp_add_cheat_layout);
+	main_layout->addWidget(grp_add_cheat);
+
+	setLayout(main_layout);
+
+	// Edit/Manage UI
+	connect(tbl_cheats, &QTableWidget::itemClicked, [=](QTableWidgetItem* item) {
+		int row           = tbl_cheats->currentRow();
```

Comment:
row can be -1 if you click on empty space

Reply:
I put a check just in case.